### PR TITLE
Fix the dump date-import of the dumpv4

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request
After merging https://github.com/meilisearch/meilisearch/pull/3012 I realized that the tests on the date of the dump-v4 were still ignored, thus, I fixed them and then noticed #3012 wasn't working properly.

## Related issue
Fixes https://github.com/meilisearch/meilisearch/issues/2987 a second time

@funilrys since you wrote most of the code you might be interested, but don't feel obligated to review this code. 
Someone from the team will double-check it works :grin:

### CODE REVIEW

**Comment by curquiza** [comment]
Thanks @irevoire for the fix! Since I don't want to miss this PR as part of the fix for #3012 I remove the `skip-changelog` label

Thanks again for following the job!

**Review by funilrys — DISMISSED** [review]

File: dump/src/reader/v4/mod.rs [hunk 1]
```diff
@@ -167,17 +167,16 @@ impl V4IndexReader {
...
-                    created_at = task.created_at()
+                    updated_at = task.processed_at()
                 }
 
+                // Once we reach the `creation_task_id` we can stop iterating on the task queue and
+                // this task represent our `created_at`.
                 if task.id as usize == index_metadata.creation_task_id {
-                    created_at = task.processed_at();
-
+                    created_at = task.created_at();
```
[the remark this answers is not in the document]
Comment (irevoire) [reply dump/src/reader/v4/mod.rs:179]: Yes, actually, this time, the index creation was done synchronously, which means the important timing was the task creation date and not the `processed` time!

File: dump/src/reader/v4/mod.rs [hunk 2]
```diff
@@ -167,17 +167,16 @@ impl V4IndexReader {
...
-                    updated_at = task.updated_at()
-                }
-
-                if created_at.is_none() {
-                    created_at = task.created_at()
+                    updated_at = task.processed_at()
                 }
 
+                // Once we reach the `creation_task_id` we can stop iterating on the task queue and
+                // this task represent our `created_at`.
```
Comment (dureuill) [inline dump/src/reader/v4/mod.rs:177]: typo

```suggestion
                // this task represents our `created_at`.
```

**Review by dureuill — CHANGES_REQUESTED** [review]
Thank you for the changes!

I know this is not something coming from this PR, but I'm wondering about the possible 2 calls to `now_utc` [on this line and the next](https://github.com/meilisearch/meilisearch/pull/3190/files#diff-5ca2c8d1bf39067c35add46adc13a9774d0442bd1bf597770a86f6deaa746a28R188).

This makes it so it is possible to have two distinct times, is that a problem?

If so we could hoist the call to a local variable and pass the resulting offset to both fields that might need it. IMO it is OK to make the call even if the value is not ultimately needed because we're not in a performance-sensitive part of the application (if you feel otherwise you could guard the call on having one of created_at or updated_at not be None).

If having two distinct times is intended, maybe we should specify so in a comment here?

What do you think?

**Review by dureuill — APPROVED** [review]
Thank you! Looks good to me!

**Comment by irevoire** [comment]
bors merge

**Comment by amab8901** [comment]
Why do I keep getting these notifications all the time? 
image

**Pull request merged**